satpy/composites/__init__.py

```diff
@@ -309,10 +312,28 @@ def apply_modifier_info(self, origin, destination):
                 elif o.get(k) is not None:
                     d[k] = o[k]
 
+    def match_data_arrays(self, data_arrays):
+        """Match data arrays so that they can be used together in a composite."""
+        self.check_areas(data_arrays)
+        return self.drop_coordinates(data_arrays)
+
+    def drop_coordinates(self, data_arrays):
+        """Drop neglible non-dimensional coordinates."""
+        new_arrays = []
+        for ds in data_arrays:
+            drop = [coord for coord in ds.coords
+                    if coord not in ds.dims and any([neglible in coord for neglible in NEGLIBLE_COORDS])]
+            if drop:
+                new_arrays.append(ds.drop(drop))
+            else:
+                new_arrays.append(ds)
+
+        return new_arrays
+
     def check_areas(self, data_arrays):
         """Check that the areas of the *data_arrays* are compatible."""
         if len(data_arrays) == 1:
-            return data_arrays
+            return
```

Comment:
These return values in `check_areas` need to be put back unless we are breaking backwards compatibility for everyone who has a custom composite outside of Satpy.

Reply:
how about we rename `check_areas` to `_check_areas`, and make a new `check_areas` that calls `match_data_arrays` and issues a deprecation warning ?

Reply:
I'd be ok with that. Or, given how we may move away from area definition in the long term, call it something like `check_geolocation` or `check_coordinates`?

Reply:
Fine by me (you mean instead of `_check_areas` right ?)

Reply:
Yes. So `check_areas` calls `check_geolocation` and `match_data_arrays` calls `check_geolocation`.

Reply:
hmm, then we need to reintroduce the return values. I was suggesting the new `check_areas` would call `match_data_arrays`...

Reply:
🤦‍♂ Oh yeah, duh. That makes sense. So `match_data_arrays` calls `check_geolocation` and `drop_coordinates`. `check_areas` calls `match_data_arrays` with a deprecation warning.